### PR TITLE
[Refactor] Classification 3/n

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Continues work on classification refactor https://github.com/Lightning-AI/metrics/issues/1001
Prior work: 
* https://github.com/Lightning-AI/metrics/pull/1143
* https://github.com/Lightning-AI/metrics/pull/1054

This PR takes care of `precision`, `recall`, `fbeta` and `f1` metrics which all are subclasses from the stat scores metric. Smaller changes to other files as I found a couple of mistakes from the prior work.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/classification/f_beta.py [hunk 1]
```diff
@@ -13,13 +13,705 @@
...
+        beta: Weighting between precision and recall in calculation. Setting to 1 corresponds to equal weight
+        num_classes: Integer specifing the number of classes
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``micro``: Sum statistics over all labels
+            - ``macro``: Calculate statistics for each label and average them
+            - ``weighted``: Calculates statistics for each label and computes weighted average using their support
+            - ``"none"`` or ``None``: Calculates statistic for each label and applies no reduction
+        top_k:
```
Comment (Borda) [inline src/torchmetrics/classification/f_beta.py:160]:
```suggestion
        top_k:

```

File: src/torchmetrics/classification/f_beta.py [hunk 2]
```diff
@@ -13,13 +13,705 @@
...
+
+        ignore_index:
+            Specifies a target value that is ignored and does not contribute to the metric calculation
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
+            Set to ``False`` for faster computations.
+
+    Returns:
+        The returned shape depends on the ``average`` and ``multidim_average`` arguments:
+
+        - If ``multidim_average`` is set to ``global``
```
Comment (Borda) [inline src/torchmetrics/classification/f_beta.py:178]:
```suggestion
        - If ``multidim_average`` is set to ``global``:
```

File: src/torchmetrics/classification/f_beta.py [hunk 3]
```diff
@@ -13,13 +13,705 @@
...
+
+          - If ``average='micro'/'macro'/'weighted'``, the output will be a scalar tensor
+          - If ``average=None/'none'``, the shape will be ``(C,)``
+
+        - If ``multidim_average`` is set to ``samplewise``
```
Comment (Borda) [inline src/torchmetrics/classification/f_beta.py:183]:
```suggestion
        - If ``multidim_average`` is set to ``samplewise``:

```

File: src/torchmetrics/functional/classification/precision_recall.py [hunk 4]
```diff
@@ -15,11 +15,641 @@
...
+        target: Tensor with true labels
+        num_classes: Integer specifing the number of classes
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``micro``: Sum statistics over all labels
+            - ``macro``: Calculate statistics for each label and average them
+            - ``weighted``: Calculates statistics for each label and computes weighted average using their support
+            - ``"none"`` or ``None``: Calculates statistic for each label and applies no reduction
+        top_k:
```
Comment (Borda) [inline src/torchmetrics/functional/classification/precision_recall.py:185]:
```suggestion

        top_k:
```

File: docs/source/classification/fbeta_score.rst [hunk 5]
```diff
@@ -1,22 +1,64 @@
...
+.. autofunction:: torchmetrics.functional.binary_fbeta_score
+    :noindex:
+
+multiclass_fbeta_score
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. autofunction:: torchmetrics.functional.multiclass_fbeta_score
+    :noindex:
+
+multilabel_fbeta_score
```
Comment (Borda) [inline docs/source/classification/fbeta_score.rst:60]: Isn't it the same as `multiclass_fbeta_score`, just that the label sum per sample is exactly one?

File: tests/unittests/classification/test_jaccard.py [hunk 6]
```diff
@@ -128,11 +125,7 @@ def _sk_jaccard_index_multiclass(preds, target, ignore_index=None, average="macr
         preds = np.argmax(preds, axis=1)
     preds = preds.flatten()
     target = target.flatten()
-
-    if ignore_index is not None:
-        idx = target == ignore_index
-        target = target[~idx]
-        preds = preds[~idx]
+    target, preds = remove_ignore_index(target, preds, ignore_index)
```
Comment (Borda) [inline tests/unittests/classification/test_jaccard.py:128]: did not know we had this code duplication so often :D

**Review by Borda — APPROVED** [review]
I did not go into too many details but overall LGTH, just check the failing tests :)

**Review by justusschock — APPROVED** [review]

**Pull request merged**